x-pack/test/functional/page_objects/grok_debugger_page.ts

```diff
@@ -4,15 +4,91 @@
  * 2.0; you may not use this file except in compliance with the Elastic License
  * 2.0.
  */
+import expect from '@kbn/expect';
+import { FtrProviderContext } from '../ftr_provider_context';
 
-import { FtrService } from '../ftr_provider_context';
+export function GrokDebuggerPageProvider({ getPageObjects, getService }: FtrProviderContext) {
+  const testSubjects = getService('testSubjects');
+  const aceEditor = getService('aceEditor');
+  const retry = getService('retry');
+  const log = getService('log');
+  return {
+    async simulateButton() {
+      return await testSubjects.find('btnSimulate');
+    },
 
-export class GrokDebuggerPageObject extends FtrService {
-  private readonly common = this.ctx.getPageObject('common');
-  private readonly grokDebugger = this.ctx.getService('grokDebugger');
+    async getEventOutput() {
+      return await aceEditor.getValue(
+        'grokDebuggerContainer > aceEventOutput > codeEditorContainer'
+      );
+    },
 
-  async gotoGrokDebugger() {
-    await this.common.navigateToApp('grokDebugger');
-    await this.grokDebugger.assertExists();
-  }
+    async setEventInput(value: string) {
+      await aceEditor.setValue(
+        'grokDebuggerContainer > aceEventInput > codeEditorContainer',
+        value
+      );
+    },
+
+    async setPatternInput(pattern: string) {
+      await aceEditor.setValue(
+        'grokDebuggerContainer > acePatternInput > codeEditorContainer',
+        pattern
+      );
+    },
+
+    async setCustomPatternInput(customPattern: string) {
+      await aceEditor.setValue(
+        'grokDebuggerContainer > aceCustomPatternsInput > codeEditorContainer',
+        customPattern
+      );
+    },
+
+    async toggleSetCustomPattern() {
+      await testSubjects.click('grokDebuggerContainer > btnToggleCustomPatternsInput');
+    },
+
+    async executeGrokSimulation(input: string, pattern: string, customPattern: string | null) {
+      let value;
+      await this.setEventInput(input);
+      await this.setPatternInput(pattern);
+      if (customPattern) {
+        await this.toggleSetCustomPattern();
+        await this.setCustomPatternInput(customPattern);
+      }
+      await (await this.simulateButton()).click();
+      await retry.try(async () => {
+        log.debug(await this.getEventOutput());
+        value = JSON.parse(await this.getEventOutput());
+        log.debug(Object.keys(value));
+        expect(Object.keys(value).length).to.be.greaterThan(0);
+      });
+      return value;
+    },
+
+    // This needs to be fixed to work with the new test functionality. This method is skipped currently.
```

Comment:
Similar to my comment in `grok_debugger/home_page.ts`, I'd like to get a better understanding of what is needed to be fixed here and whether it can be addressed as part of this PR.